src/classes/EDAHealthCheckGroupAPIService.cls

```diff
@@ -0,0 +1,92 @@
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Health Check
+* @group-content ../../ApexDocContent/HealthCheck.htm
+* @description API service class for EDA HealthCheckGroup to retrieve HealthCheckGroup view models
+* corresponding to specified HealthCheckDefinition models.
+*/
+public virtual with sharing class EDAHealthCheckGroupAPIService implements HealthCheckGroupAPIServiceInterface {
+
+    /*********************************************
+    * @description Instance for Singleton Pattern
+    **********************************************/
+    @TestVisible
+    private static EDAHealthCheckGroupAPIService instance;
+
+    /*********************************************************************
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************/
+    public EDAHealthCheckGroupAPIService() {}
+
+    /******************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of HealthCheckGroupAPIServiceInterface.
+    ******************************************************************************/
+    public static HealthCheckGroupAPIServiceInterface getInstance() {
+        if (instance == null) {
+            instance = new EDAHealthCheckGroupAPIService();
+        }
+
+        return instance;
+    }
+
+    /******************************************************************************************
+    * @description Retrieves a view model representing a health check group for a specified 
+    * HealthCheckDefinition.
+    * @param  name The name of a HealthCheckDefinition
+    * @return A HealthCheckGroupVModel representing a specified HealthCheckDefinition; null if one
+    * matching the name provided does not exist.
+    ******************************************************************************************/
+    public virtual HealthCheckGroupVModel getHealthCheckGroup(String name){
+        String caseInsensitiveName = name.toLowerCase();
+
+        List<HealthCheckItemVModel> hlthChkItemsList = new List<HealthCheckItemVModel>();
+        List<String> rowKeysList = new List<String>();
+
+        switch on caseInsensitiveName {
+            when 'reciprocalrelationship' {
```

Comment:
should we consider to use final upper case variable value for this?

Reply:
Since we are using case insensitivity by lowercasing this earlier, I feel like lowercase is an appropriate call here.

Reply:
Since these are strings coming in from a different source I'd like to keep it this way for now. Once we settle on the interoperability solution we should circle back and add some form of more permanent resolution.